SCClassLibrary/Common/Core/Condition.sc

```diff
@@ -16,21 +16,31 @@ Condition {
 		value.yield;
 	}
 
+	setTimeout { |timeout, action|
+		var waitingThread = thisThread.threadPlayer;
+		var timeoutThread = Routine {
+			timeout.wait;
+			waitingThreads.remove(waitingThread);
+			waitingTimeouts.remove(timeoutThread);
+			action.value;
+			waitingThread.clock.sched(0, waitingThread);
+		};
+		waitingTimeouts = waitingTimeouts.add(timeoutThread);
+		timeoutThread.play(thisThread.clock);
+	}
+
 	signal {
-		var tempWaitingThreads, time;
-		if (test.value, {
-			time = thisThread.seconds;
-			tempWaitingThreads = waitingThreads;
-			waitingThreads = nil;
-			tempWaitingThreads.do({ arg thread;
-				thread.clock.sched(0, thread);
-			});
-		});
+		if(test.value) {
+			this.unhang;
+		};
```

Comment:
if you liked, you could remove all semicolons.

Reply:
This is one where, to be honest, I don't fully agree. Removing trailing semicolons may be slightly visually cleaner... though to me, then the last statement in a block is delimited differently from other statements, rather as if I omitted the trailing period from this paragraph

See? It raises a question for the reader... why is it missing here but required elsewhere? I know it's a C convention but I've grown to dislike it.

Also, future revisions may need to add them back.

If it's a recommendation in the style guide, I'll remove them.

Reply:
My own reasoning for omitting semicolons is: 
- when a method does just one thing 
- to signal that this is the return value of the function

For me, here it is the first.

But I don't care much, definitely up to you!

Reply:
> If it's a recommendation in the style guide, I'll remove them.

it is -- https://github.com/supercollider/supercollider/wiki/Code-style-guidelines#return-statements

please check it first next time, could have saved some discussion here :)

Reply:
OK, done. Sorry for noise.